src/main/java/assets/game.js

```diff
@@ -176,6 +184,39 @@ function place(size) {
             }
             cell.classList.toggle("placed");
         }
+        if(sub){
+            let cell;
+            if(vertical) {
+                let tableRow = table.rows[row+1];
+                if (tableRow === undefined) {
+                    // ship is over the edge; let the back end deal with it
+                    sub = false;
+                    size = 5;
+                    return;
+                }
+                cell = tableRow.cells[col+1];
+            } else {
+                let tableRow = table.rows[row-1];
+                if (tableRow === undefined) {
+                    // ship is over the edge; let the back end deal with it
+                    sub = false;
+                    size = 5;
+                    return;
+                }
+                let tmp = col+1;
+                cell = tableRow.cells[tmp];
+            }
+            if (cell === undefined) {
+                // ship is over the edge; let the back end deal with it
+                sub = false;
+                size = 5;
+                return;
+            }
+            cell.classList.toggle("placed");
+
+            sub = false;
+            size = 5;
+        }
```

Comment:
Looks good.  Uses the same ideas as the rest of the place ship function.  You handle the sub being placed over the edge of the board well.